utxonursery.go

```diff
@@ -1875,7 +1872,22 @@ func (k *kidOutput) Decode(r io.Reader) error {
 	}
 	k.witnessType = lnwallet.WitnessType(byteOrder.Uint16(scratch[:2]))
 
-	return lnwallet.ReadSignDescriptor(r, &k.signDesc)
+	if err := lnwallet.ReadSignDescriptor(r, &k.signDesc); err != nil {
+		return err
+	}
+
+	_, err = r.Read(scratch[:4])
+	switch {
+	case err == io.EOF:
+		// Old format, nothing more to read.
+		return nil
```

Comment:
This is actually a database migration. I see that this is very few lines of code, but shouldn't it be a proper migration? With migrations we have the option at some point in the future to decide to remove some of the oldest ones. We would want to do this, because they can be hard to maintain without duplicating code. This code will then probably not found back anymore.

Reply:
I don't know if this is really the same. The example of `channel.go` looks like normal logic for a variable length structure, code that can always be hit. While in this PR it is specifically addressing an old format. At some point it will never be hit anymore.

But just wanted to bring it up to make a conscious decision.

Reply:
What makes us not require a db migration in this case is that it is backwards and forwards compatible: This new code can read the the old format AND the old code can read the new format. 

A better example thatn what I gave earlier is here: https://github.com/lightningnetwork/lnd/blob/6d759bd8d9f1595f32fba6f901db4dcb78b6613e/channeldb/channel.go#L2149

Reply:
Yes, I see. I know a migration is not required functionally, but my point is just that those EOF checks could be removed if there was a migration for this change. At some point, the oldest migrations themselves will probably also be removed, eventually resulting in less code overall (no EOF check, no migration).

But I understand that in case of such a simple EOF check, you could go either way.